src/app/rpmostree-pkg-builtins.c

```diff
@@ -146,38 +178,29 @@ rpmostree_builtin_pkg_add (int            argc,
   for (int i = 1; i < argc; i++)
     {
       const char *pkgspec = argv[i];
-      g_autofree char *abspath = NULL;
 
-      if (!g_str_has_suffix (pkgspec, ".rpm") || /* repo install */
-           g_str_has_prefix (pkgspec, "/"))      /* local install */
+      if (!g_str_has_suffix (pkgspec, ".rpm")) /* repo install */
         {
           g_ptr_array_add (packages_to_add, g_strdup (pkgspec));
           continue;
         }
-
-      /* OK, it's a relative path, we need to check that it
-       * exists and canonicalize it for the daemon. */
-
-      if (access (pkgspec, R_OK) != 0)
-        {
-          glnx_set_prefix_error_from_errno (error,
-                                            "can't read package '%s': ",
-                                            pkgspec);
-          return EXIT_FAILURE;
-        }
-
-      abspath = realpath (pkgspec, NULL);
-      if (abspath == NULL)
+      else /* local RPM install */
         {
-          glnx_set_prefix_error_from_errno (error, "%s", "realpath");
-          return EXIT_FAILURE;
+          int fd = open (pkgspec, O_RDONLY);
```

Comment:
Done! (Though note that this one we actually close and the dup that gets added to the list will have `O_CLOEXEC`).